app.rb

```diff
@@ -0,0 +1,159 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def welcome_message
+    puts
+    puts "Welcome to School Library App! #{Time.now}"
+    puts
+  end
+
+  def message
+    puts
+    puts "Please choose an option by entering a number:
+	1 - List all books
+	2 - List all people
+	3 - Create a person
+	4 - Create a book
+	5 - Create a rental
+	6 - List all rentals for a given person id
+	7 - Exit
+	"
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def logic
+    number = gets.chomp.to_i
+    case number
+    when 1
+      list_all_books
+    when 2
+      list_all_people
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    when 7
+      abort('Thank you for using this App!')
+    else
+      abort('Wrong Input Range [1..7]')
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    input = gets.chomp.to_i
+    case input
+    when 1
+      helper_func_create_student
+    when 2
+      helper_func_create_teacher
+    end
+  end
+
+  def helper_func_create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Classroom (example A for section-A, B for section-B): '
+    classroom = gets.chomp
+    student = Student.new(age, classroom, name)
+    @people.push(student)
+    print 'Person created succesfully'
+    print
+    message
+    logic
+  end
+
+  def helper_func_create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    spec = gets.chomp
+    teacher = Teacher.new(age, spec, name)
+    @people.push(teacher)
+    print 'Person created successfully'
+    message
+    logic
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    print 'Book created succesfully'
+    message
+    logic
+  end
+
+  def create_rental
+    print 'Select a book from the following list by S.No'
+    puts
+    helper_func_list_all_books
+    book_input = gets.chomp.to_i
+    print 'Select a person from the following list by S.No (not id)'
+    puts
+    helper_func_list_all_people
+    person_input = gets.chomp.to_i
+    date = Time.now.strftime('%Y/%m/%d')
+    rental_created = Rental.new(date, @books[book_input - 1], @people[person_input - 1])
+    @rentals.push(rental_created)
+    print 'Retnal created succesfully'
+    message
+    logic
+  end
```

Comment:
Great job @PRATAP-KUMAR 🤝

- [ ] [Optional] Kindly make sure the create rental option behaves as expected. when I select `S.NO 1` nothing seems to happen. In other words, no rental is created but for the `S.NO 2`, the rental is created successfully. you may kindly work around this problem and find a solution.

Reply:
@1GORDON thanks for your suggestion.

I just tried with a single person to create a rental and I can see the rental generated. Maybe I am missing something from your suggestion please correct me.

```
[admin@ANTARCTICA oop-school-library]$ ruby main.rb 

Welcome to School Library App! 2022-11-02 13:05:10 +0530


Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
3
Do you want to create a student (1) or a teacher (2)? [Input the number]:1
Age: 15
Name: student
Classroom (example A for section-A, B for section-B): a
Person created succesfullya

Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
3
Do you want to create a student (1) or a teacher (2)? [Input the number]:2
Age: 35
Name: teacher
Specialization: teaching
Person created successfully
Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
4
Title: book1
Author: author1
Book created succesfully
Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
4    
Title: book2
Author: author2
Book created succesfully
Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
5
Select a book from the following list by S.No
S.No: 1 | Title: book1 By Author: author1
S.No: 2 | Title: book2 By Author: author2
1
Select a person from the following list by S.No (not id)
S.No: 1 | [Student] | Name: student | ID: 54 | Age: 15
S.No: 2 | [Teacher] | Name: teacher | ID: 21 | Age: 35
1
Retnal created succesfully
Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
6
the following persons have rented books, Select id of the person
54 | student
54
Date: 2022/11/02, Book "book1" by author1

Please choose an option by entering a number:
	1 - List all books
	2 - List all people
	3 - Create a person
	4 - Create a book
	5 - Create a rental
	6 - List all rentals for a given person id
	7 - Exit
	
7
Thank you for using this App!
[admin@ANTARCTICA oop-school-library]$ 
```